### PR TITLE
Create piece model + Individual sub-class for each piece type

### PR DESCRIPTION
**Pull request by zoheb391**
Created a piece model and individual models for each piece type that inherits the parent "piece".

Also, some additions to game model's database

### CODE REVIEW

File: db/migrate/20160310030824_create_pieces.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class CreatePieces < ActiveRecord::Migration
+  def change
+    create_table :pieces do |t|
+    	t.integer :x_coordinate
+    	t.integer :y_coordinate
+    	t.string :color
+    	t.boolean :taken
+    	t.integer :player_id
```
Comment (ilyakrasnov) [inline db/migrate/20160310030824_create_pieces.rb:8]: Probably no need to connect the player with the piece. Could do it via the color.

File: db/migrate/20160306052733_create_games.rb [hunk 2]
```diff
@@ -4,7 +4,9 @@ def change
     	t.integer :white_player_id
     	t.integer :black_player_id
     	t.integer :winning_player_id
-    	t.string :turn
+    	t.integer :turn
+    	t.integer :game_id
+    	t.string :game_name
```
Comment (ilyakrasnov) [inline db/migrate/20160306052733_create_games.rb:9]: Does it make sense to have a "game_name" as opposed to "name"?

File: db/migrate/20160310030824_create_pieces.rb (hunk 1 above)
Comment (AncyentMariner) [inline db/migrate/20160310030824_create_pieces.rb:8]: So would it be best to remove the player_id column in this table, then?

Comment (zoheb391) [inline db/migrate/20160310030824_create_pieces.rb:8]: Ahh, makes sense. I'll make the changes

**Pull request closed without merging**